core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1399,8 +1399,20 @@ private[spark] class DAGScheduler(
           case _ =>
             updateAccumulators(event)
         }
-      case _: ExceptionFailure | _: TaskKilled => updateAccumulators(event)
-      case _ =>
+
+      case fail =>
+        if (task.metrics.shuffleReadMetrics.recordsRead == 0) {
+          val reduceId = task.partitionId
+          val shuffleReadSize = stage.parents.filter(_.isInstanceOf[ShuffleMapStage])
+            .map(_.asInstanceOf[ShuffleMapStage].shuffleDep.shuffleId)
+            .map(mapOutputTracker.shuffleStatuses.get(_)).map(_.get.mapStatuses)
+            .flatMap(_.map(_.getSizeForBlock(reduceId))).sum
+          task.metrics.shuffleReadMetrics.setRecordsRead(shuffleReadSize)
```

Comment:
This is not the total records that have been read, why set it to the metrics?

Reply:
Thanks for you reply.
I have changed the code and it only set shuffle read size for ExecutorLostFailure.

 I think the reason for executorLostFailure always is data skew.
So the shuffle read size is important for a user, but now it always shown as 0 when ExecutorLostFailure occurred.